src/debug.c

```diff
@@ -54,6 +54,10 @@ typedef ucontext_t sigcontext_t;
 #endif
 #endif
 
+#if defined(__APPLE__) && defined(__arm64__) && defined(USE_JEMALLOC)
```

Comment:
@dgilman what does this have to do with jemalloc? i suppose that part should be dropped.

@Developer-Ecosystem-Engineering maybe you can explain how #7453 didn't add that include, and if there's something else we're missing here...

Reply:
It may not hurt to drop it but it is only required for the jemalloc builds. It wasn't caught on the initial patch for mac arm64 because redis on mac is compiled with system malloc by default so it seems reasonable that everyone forgot to test the patch with jemalloc on mac.

Reply:
but why isn't it needed? `arm_thread_state64_get_pc` is called regardless of which allocator is being used.
maybe as a side effect of using jemalloc this include finds it's way in, and then everything works.
but if that's the case, we should not rely on that.

Reply:
The #include for mach/mach.h was probably happening by default on the non jemalloc build and not in the jemalloc build. It wasn't added as we were able to build & test without it